src/SpectralDataDialog.cpp

```diff
@@ -0,0 +1,251 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  SpectralDataDialog.cpp
+
+  Edward Hui
+
+*******************************************************************//*!
+
+\class SpectralDataDialog
+\brief Provides UI for spectral editing and parameters adjustments
+
+*//*******************************************************************/
+
+
+#include "SpectralDataDialog.h"
+
+#include <wx/app.h>
+#include <wx/defs.h>
+#include <wx/button.h>
+#include <wx/dialog.h>
+#include <wx/event.h>
+#include <wx/frame.h>
+#include <wx/imaglist.h>
+#include <wx/intl.h>
+#include <wx/listctrl.h>
+#include <wx/settings.h>
+#include <wx/spinctrl.h>
+#include <wx/stattext.h>
+#include <wx/textctrl.h>
+
+#include "../images/Arrow.xpm"
+#include "../images/Empty9x16.xpm"
+
+#include "AudioIO.h"
+#include "Clipboard.h"
+#include "UndoManager.h"
+#include "Project.h"
+#include "ProjectFileIO.h"
+#include "ProjectHistory.h"
+#include "ProjectSettings.h"
+#include "ShuttleGui.h"
+#include "SpectralDataManager.h"
+#include "TrackPanel.h"
+#include "widgets/AudacityMessageBox.h"
+
+enum {
+   ID_ON_APPLY = 10000,
+   ID_CHECKBOX_SMART,
+   ID_CHECKBOX_OVERTONES,
+   ID_SLIDER_BRUSH_SIZE
+};
+
+BEGIN_EVENT_TABLE(SpectralDataDialog, wxDialogWrapper)
+   EVT_BUTTON(ID_ON_APPLY, SpectralDataDialog::OnApply)
+   EVT_SHOW(SpectralDataDialog::OnShow)
+   EVT_CLOSE(SpectralDataDialog::OnCloseWindow)
+   EVT_SLIDER(ID_SLIDER_BRUSH_SIZE, SpectralDataDialog::OnBrushSizeSlider)
+END_EVENT_TABLE()
+
+#define Title XO("Spectral Data Control Panel")
+
+SpectralDataDialog::SpectralDataDialog(AudacityProject *parent):
+      wxDialogWrapper(FindProjectFrame( parent ), wxID_ANY, Title,
+                      wxDefaultPosition, wxDefaultSize,
+                      wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER )
+{
+   SetName();
+
+   mProject = parent;
+   mSelected = 0;
+   mAudioIOBusy = false;
```

Comment:
I prefer in-line initializers in the class, or member initializers before the function body, instead of assignments in the body, where possible.  It makes it a bit easier to compare the constructor and the class and be sure everything is initialized.

In fact, when a class has const or reference data members, those are the only ways you can initialize them, and it's good when possible to make such data members to help the reader reason about the class, knowing the members are immutable.

So are any of these immutable?  I think mProject can have type `AudacityProject *const`